src/ephemeris.c

```diff
@@ -81,28 +33,23 @@ void ephemeris_new(ephemeris_t *e)
   char buf[SID_STR_LEN_MAX];
   sid_to_string(buf, sizeof(buf), e->sid);
 
-  gps_time_t t = get_current_time();
+  if (!e->valid) {
+    log_error("Invalid ephemeris for %s", buf);
+    return;
+  }
+
   u32 index = sid_to_global_index(e->sid);
-  if (!ephemeris_valid(&es[index], &t)) {
```

Comment:
If the ephemeris in the NDB is invalid, we should store the incoming ephemeris to both the NDB and the candidate immediately. Only when the NDB already contains valid data should we wait for the candidate match before updating.